collector/test/HostHeuristicsTest.cpp

```diff
@@ -52,33 +52,18 @@ class MockHostInfoHeuristics : public HostInfo {
   MOCK_METHOD0(GetDistro, std::string&());
 };
 
-// Note that in every test below, ProcessHostHeuristics will be called first
-// with creation of a config mock, which is somewhat annoying, but doesn't
-// cause any serious issues.
-class MockCollectorConfig : public CollectorConfig {
- public:
-  MockCollectorConfig(CollectorArgs* collectorArgs)
-      : CollectorConfig(collectorArgs){};
-
-  MOCK_CONST_METHOD0(UseEbpf, bool());
-
-  void SetCollectionMethod(CollectionMethod cm) {
-    if (host_config_.HasCollectionMethod()) {
-      host_config_.SetCollectionMethod(cm);
-    }
-    collection_method_ = cm;
-  }
-};
-
 TEST(HostHeuristicsTest, TestS390XRHEL84) {
   MockS390xHeuristics s390xHeuristics;
   MockHostInfoHeuristics host;
   KernelVersion version = KernelVersion("4.18.0-305.88.1.el8_4.s390x", "", "s390x");
   CollectorArgs* args = CollectorArgs::getInstance();
-  MockCollectorConfig config(args);
+  int argc = 3;
+  const char* argv[] = {"collector", "--collection-method", "ebpf"};
```

Comment:
The collection method has to be set here, because `CollectorConfig config(args)` results in a call to `ProcessHostHeuristics` and in turn `Process`. If the collection method is set to the default `CORE_BPF` in  `Process` a fatal error will be triggered.

Reply:
Even if it's better than using a mock, it should be refactored to be more reusable.

Reply:
I am now using the mock again.